wdtk-wikibaseapi/src/main/java/org/wikidata/wdtk/wikibaseapi/ApiConnection.java

```diff
@@ -199,23 +208,90 @@ public void checkCredentials() throws IOException, MediaWikiApiErrorException {
 	 *
 	 * @return name of the logged in user
 	 */
+	@JsonProperty("username")
 	public String getCurrentUser() {
-		return this.username;
+		return username;
+	}
+
+	/**
+	 * Returns the map of tokens (such as csrf token and login token) currently used in this connection.
+	 */
+	public Map<String, String> getTokens() {
+		return Collections.unmodifiableMap(tokens);
+	}
+
+	/**
+	 * Sets the maximum time to wait for when establishing a connection, in milliseconds.
+	 * For negative values, no timeout is set.
+	 *
+	 * @see HttpURLConnection#setConnectTimeout
+	 */
+	public void setConnectTimeout(int timeout) {
+		connectTimeout = timeout;
+		if (timeout >= 0) {
```

Comment:
I'm a bit scared by this `if`. What about the following (quite bad...) use case:
```
client.setConnectTimeout(100);
client.setConnectTimeout(0);
```

I guess that, with the current code, the timeout would stay at `100`.

Same question for the read timeout.

Reply:
The following commit should fix this.